<launch>
    
  <arg name="camera_name" default="/stereo_down"/>
  <arg name="decimation" default="1" />
  <arg name="disparity_params" default="" />
  <arg name="nodelet_args" value="standalone image_proc/crop_decimate" />

  <param name="/use_sim_time" value="true"/>

  <node pkg="nodelet" type="nodelet" name="image_proc_decimate_left_x$(arg decimation)" ns="$(arg camera_name)" output="screen" respawn="true" args="$(arg nodelet_args)" >
      <param name="decimation_x" value="$(arg decimation)" />
      <param name="decimation_y" value="$(arg decimation)" />
      <remap from="camera" to="left" />
      <remap from="camera_out" to="scaled_x$(arg decimation)/left"/>
  </node>

  <node pkg="nodelet" type="nodelet" name="image_proc_decimate_right_x$(arg decimation)" ns="$(arg camera_name)" output="screen" respawn="true" args="$(arg nodelet_args)" >
    <param name="decimation_x" value="$(arg decimation)" />
    <param name="decimation_y" value="$(arg decimation)" />
    <remap from="camera" to="right" />
    <remap from="camera_out" to="scaled_x$(arg decimation)/right"/>
  </node>
  
  <node pkg="stereo_image_proc" type="stereo_image_proc" name="image_proc_left" respawn="true" output="screen">
    <remap from="left/image_raw" to="$(arg camera_name)/scaled_x$(arg decimation)/left/image_raw"/>
    <remap from="left/camera_info" to="$(arg camera_name)/scaled_x$(arg decimation)/left/camera_info"/>
    <remap from="left/image_rect_color" to="$(arg camera_name)/scaled_x$(arg decimation)/left/image_rect_color"/>
    <remap from="right/image_raw" to="$(arg camera_name)/scaled_x$(arg decimation)/right/image_raw"/>
    <remap from="right/camera_info" to="$(arg camera_name)/scaled_x$(arg decimation)/right/camera_info"/>
    <remap from="right/image_rect_color" to="$(arg camera_name)/scaled_x$(arg decimation)/right/image_rect_color"/>
    <remap from="disparity" to="$(arg camera_name)/scaled_x$(arg decimation)/disparity"/>
    <remap from="points2" to="$(arg camera_name)/scaled_x$(arg decimation)/points2"/>
    <rosparam file="$(arg disparity_params)"/>
    <param name="approximate_sync" value="true" />
  </node>

</launch>
